docs/release_notes.rst

```diff
@@ -2,12 +2,80 @@
 PUDL Release Notes
 =======================================================================================
 
-.. _release-v2023.XX.XX:
-
 ---------------------------------------------------------------------------------------
 v2023.XX.XX
 ---------------------------------------------------------------------------------------
 
+Dagster Adoption
```

Comment:
This section contains a mixture of changes and deprecations. Should I move the deprecations into the Deprecations section of the release notes? @zaneselvans

Reply:
I think using the Deprecations section sounds like a good idea.

Reply:
Sounds good. I just moves some of these change descriptions into the deprecations section.